### PR TITLE
google-oauth

### PR DESCRIPTION
**Pull request by khungking909**
## Related Tickets
- [#74705](https://edu-redmine.sun-asterisk.vn/issues/74705)

## WHAT (optional)
- Đăng nhập bằng tài khoản google

## HOW
- Thư viện làm hết
## Evidence (Screenshot or Video)
image

https://github.com/awesome-academy/dn_minaitei_2_ecommerce/assets/93854026/fad9377f-98cc-47aa-80a1-24f4c077de27

### CODE REVIEW

**Comment by khungking909** [comment]
ready

File: app/models/account.rb [hunk 1]
```diff
@@ -3,8 +3,14 @@
 class Account < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  attr_accessor :is_login_with_google
+
+  after_commit :send_mail_if_login_with_google, if: -> { is_login_with_google }
```
Comment (binhpt-3177) [inline app/models/account.rb]:
```suggestion
  after_commit :send_mail_if_login_with_google, if: :is_login_with_google
```

File: app/models/account.rb [hunk 2]
```diff
@@ -13,4 +19,18 @@ class Account < ApplicationRecord
...
 
   validates :name, presence: true, length: { maximum: Settings.DIGIT_50 }
+
+  def self.from_omniauth(auth)
+    where(email: auth.info.email).first_or_create do |account|
+      account.email = auth.info.email
+      account.is_login_with_google = true
+      account.password = SecureRandom.hex(5)
+      account.name = auth.info.name
+      account.confirmed_at = Time.zone.now
```
Comment (binhpt-3177) [inline app/models/account.rb]:
```suggestion
      account.confirmed_at = Time.current
```

**Review by binhpt-3177 — COMMENTED** [review]
Fix conflic

**Comment by khungking909** [comment]
ready

**Review by binhpt-3177 — APPROVED** [review]

**Pull request merged**